src/neuroconv/tools/nwb_helpers/_metadata_and_file_helpers.py

```diff
@@ -370,3 +375,51 @@ def configure_and_write_nwbfile(
 
     with IO(output_filepath, mode="w") as io:
         io.write(nwbfile)
+
+
+def configure_and_export_nwbfile(
+    nwbfile: NWBFile,
+    export_nwbfile_path: Path,
+    backend_configuration: BackendConfiguration,
+) -> None:
+    configure_backend(nwbfile=nwbfile, backend_configuration=backend_configuration)
+
+    IO = BACKEND_NWB_IO[backend_configuration.backend]
+    nwbfile.set_modified()
+    with IO(export_nwbfile_path, mode="w") as io:
+        io.export(nwbfile=nwbfile, src_io=nwbfile.read_io, write_args=dict(link_data=False))
+
+
+def repack_nwbfile(
+    *,
+    nwbfile: NWBFile,
```

Comment:
I think the signature of this function is intended to work on an input NWB file path (some `.nwb` file that has already been written to disk, albeit with non-ideal chunking/compression)

That's also why I suggested started by drafting test structure first to show how you 'expect' the API for the function to feel like before deciding on implementation strategy (though I am aware that both can inform each other in an iterative review process)

Basically, I think our goal is to make something analogous to `h5repack` but much more performant and with much better default behavior by setting more optimal chunk shapes and compression methods

Reply:
Yes, that is what we want, but I do see an advantage of creating this API first and having the file reading in a separate function. Then we can do this reading from disk or reading from cloud without changing the internal logic of this function at all.

Reply:
I'm thinking something like:

```python
def repack_nwbfile(
    filepath_in: str,
    filepath_out: str
    backend_in: Literal[None, "hdf5", "zarr"] = None,  # if None, infer backend
    backend_out: Literal[None, "hdf5", "zarr"] = None,  # if None, infer from backend_configuration_changes. if that is None, use same as input
    use_default_configuration=True,  # if False, use existing backend
    backend_configuration_changes=None,
)
```

and then it would be great to also have a CLI for this. The backend_configuration_changes might be hard to specify with a CLI but we could try to do something analogous to h5repack

Reply:
Wouldn't we always want to infer the `backend_in`?

Reply:
yeah, probably. I was thinking there might be edge cases where it might be difficult to infer or where supplying it will allow us to skip the inference step, which might speed this up, but I think currently our backend inference should be able to work very fast and accurately so for now I think it would be fine to not include this